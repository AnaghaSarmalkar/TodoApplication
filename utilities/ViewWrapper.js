import React from "react";
import { View } from "react-native";

export default function ViewWrapper(props) {
  return (
    <View
      {...props}
      // style={{
      //   flex: 1,
      //   alignItems: "center",
      //   justifyContent: "center",
      //   // paddingTop: insets.top,
      // }}
    >
      {props.children}
    </View>
  );
}
