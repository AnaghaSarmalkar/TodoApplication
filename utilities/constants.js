import { Platform } from "react-native";

export const isIos = Platform.OS === "ios";
export const isWeb = Platform.OS === "web";
export const isAndroid = Platform.OS === "android";
