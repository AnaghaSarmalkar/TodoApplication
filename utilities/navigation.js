export function setPageTitle(navigation, title) {
  if (title) {
    navigation.setOptions({
      title: title,
    });
  }
}
