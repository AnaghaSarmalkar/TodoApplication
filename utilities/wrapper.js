import React from "react";
import { SafeAreaView } from "react-native-safe-area-context";

function SafeAreaWrapper({ children }) {
  <SafeAreaView>{children}</SafeAreaView>;
}
