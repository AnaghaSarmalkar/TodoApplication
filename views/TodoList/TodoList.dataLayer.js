import { gql, useQuery } from "@apollo/client";
import React, { useEffect } from "react";
import { useSafeAreaInsets } from "react-native-safe-area-context";
import TodoList from "./TodoList";

export const getLocationsSimple = gql`
  query GetLocationsSimple {
    locations {
      id
      name
    }
  }
`;

export default function TodoListDataLayer({ navigation, route }) {
  const insets = useSafeAreaInsets();
  const { loading, error, data } = useQuery(getLocationsSimple);

  const onPress = (id) =>
    navigation.navigate("Todo", {
      id: id,
    });

  useEffect(() => {
    navigation.setOptions({
      title: "LOCATIONS",
    });
  });

  return (
    <TodoList
      data={data}
      loading={loading}
      error={error}
      insets={insets}
      onItemPress={onPress}
    />
  );
}
