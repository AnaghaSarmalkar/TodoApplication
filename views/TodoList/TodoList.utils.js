export function onPressTodo(navigation, todoId) {
  if (todoId) {
    navigation.navigate("Todo", {
      id: todoId,
    });
  }
}

export function setPageTitle(navigation, title) {
  if (title) {
    navigation.setOptions({
      title: title,
    });
  }
}
