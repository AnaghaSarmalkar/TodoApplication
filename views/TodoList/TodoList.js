import React from "react";
import { View } from "react-native";
import Error from "../../components/Error/Error";
import Header from "../../components/Header/Header";
import Item from "../../components/Item/Item";
import List from "../../components/List/List";
import Loading from "../../components/Loading/Loading";
import { isWeb } from "../../utilities/constants";

export default function TodoList({
  data,
  loading,
  error,
  insets,
  onItemPress,
}) {
  if (loading)
    return (
      <View
        style={{
          flex: 1,
          alignItems: "center",
          justifyContent: "center",
          paddingTop: insets.top,
        }}
      >
        <Loading />
      </View>
    );
  if (error)
    return (
      <View
        style={{
          flex: 1,
          alignItems: "center",
          justifyContent: "center",
          paddingTop: insets.top,
        }}
      >
        <Error message={error.message} />
      </View>
    );
  const header = isWeb ? <Header title="Todo List" /> : null;

  const renderItem = ({ item }) => (
    <Item item={item} onItemPress={onItemPress} />
  );
  return (
    <View
      style={{
        flex: 1,
        alignItems: "center",
        justifyContent: "center",
        paddingTop: insets.top,
      }}
    >
      <List
        data={data.locations}
        headerComponent={header}
        renderItem={renderItem}
      />
    </View>
  );
}
