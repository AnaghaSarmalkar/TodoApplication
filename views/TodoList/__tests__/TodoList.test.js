import React from "react";
import { shallow } from "../../../utilities/testing";
import TodoList from "../TodoList";

describe("TodoList", () => {
  it("renders loading component", async () => {
    const wrapper = shallow(
      <TodoList
        data={undefined}
        loading={true}
        error={undefined}
        insets={undefined}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders error component", async () => {
    const error = new Error("An error occured.");
    const wrapper = shallow(
      <TodoList
        data={undefined}
        loading={false}
        error={error}
        insets={undefined}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders List component", async () => {
    const data = {
      data: {
        locations: [
          { id: "1", name: "Test Location 1" },
          { id: "2", name: "Test Location 2" },
          { id: "3", name: "Test Location 3" },
        ],
      },
    };

    const wrapper = shallow(
      <TodoList
        data={data}
        loading={false}
        error={undefined}
        insets={undefined}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
