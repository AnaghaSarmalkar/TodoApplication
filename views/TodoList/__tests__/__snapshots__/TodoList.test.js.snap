// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders List component 1`] = `
<ForwardRef>
  <List
    headerComponent={
      <Header
        title="Todo List"
      />
    }
    renderItem={[Function]}
  />
</ForwardRef>
`;

exports[`TodoList renders error component 1`] = `
<Error
  message="An error occured."
/>
`;

exports[`TodoList renders loading component 1`] = `<Loading />`;
